CRM/Contribute/Form/AbstractEditPayment.php

```diff
@@ -249,6 +249,14 @@ public function buildValuesAndAssignOnline_Note_Type($id, &$values) {
     // Also don't allow user to update some fields for recurring contributions.
     if (!$this->_online) {
       $this->_online = CRM_Utils_Array::value('contribution_recur_id', $values);
+      // Make additional Details editable if payment processor supports backoffice edit
+      if ($this->_online) {
+        $backOfficePaymentProcessors = CRM_Contribute_BAO_ContributionRecur::getBackOfficePaymentProcessors();
```

Comment:
Per prev by the time you get to this line 

    $this->assignPaymentRelatedVariables();

has been called - so you just need to check the array to find out if any recurring processors are valid for backoffice

$this->_recurPaymentProcessors